Add FAVITES-Lite v1.0.1 and its dependencies

niemasd opened the pull request:
This PR adds [FAVITES-Lite v1.0.1](https://github.com/niemasd/FAVITES-Lite/releases/tag/1.0.1) as well as its dependencies:

* ~*Update CoaTran from [v0.0.1](https://github.com/bioconda/bioconda-recipes/blob/ccdeb5760ceb59e1db08a52f61cdfb225b518306/recipes/coatran/meta.yaml#L2) to [v0.0.4](https://github.com/niemasd/CoaTran/releases/tag/0.0.4)*~
    * I moved this into a new PR: https://github.com/bioconda/bioconda-recipes/pull/46308
* Add [GEMF_FAVITES v1.0.3](https://github.com/niemasd/GEMF/releases/tag/1.0.3)
* Add [NiemaGraphGen v1.0.6](https://github.com/niemasd/NiemaGraphGen/releases/tag/1.0.6)
* Add [TreeSAP v1.0.10](https://pypi.org/project/treesap/1.0.10/)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

niemasd commented:
@BiocondaBot please add label

niemasd commented:
@bioconda/core This PR is ready to review; wasn't sure if you wanted me to tag you as per:

> - PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
> - Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or `ping @bioconda/core` in a comment.

bgruening reviewed (commented):
The identifier goes into an extra section for example together with the maintainer. The recipe which you modified had this for example. Can you please create an extra PR for the recipe that you update. Thanks!

niemasd commented:
@bgruening Thank you for reviewing! I have made the following edits to this PR:

* I reverted all of my edits to the existing tool's recipe (`coatran`)
    * I opened a separate PR with just that single tool's update
        * https://github.com/bioconda/bioconda-recipes/pull/46308
* Based on your feedback about an `extra` section with `identifiers` and `maintainers`, I also updated `meta.yaml` for 2 existing tools I contributed to add the `extra` section, and created 2 separate PRs
    * https://github.com/bioconda/bioconda-recipes/pull/46310
    * https://github.com/bioconda/bioconda-recipes/pull/46311

Now, this PR only has new additions (which can be confirmed from the "Files changed" tab: all are new files). Can you check again?

niemasd commented:
@bioconda/core This PR is ready to review

niemasd commented:
@bgruening @mencian Can you review this one? I updated it based on @bgruening's feedback; hoping to get these tools incorporated into bioconda soon 😄

bgruening reviewed (approved):
Very cool! Very good quality. Thanks!
